src/Common/src/Internal/Cryptography/BasicSymmetricCipherBCrypt.cs

```diff
@@ -30,9 +30,9 @@ public BasicSymmetricCipherBCrypt(SafeAlgorithmHandle algorithm, CipherMode ciph
 
             _hKey = algorithm.BCryptImportKey(key);
 
-            if (effectiveKeyLength != 0)
+            if (ownsParentHandle)
             {
-                Cng.SetEffectiveKeyLength(_hKey, effectiveKeyLength);
+                _hKey.SetParentHandle(algorithm);
```

Comment:
>> ...but doesn't cause the algorithm handle to Dispose (it ends up going off to the finalizer).

Not sure I follow. When BasicSymmetricCipherBCrypt is disposed, it will dispose _hKey which will call _parentHandle.DangerousRelease (where _parentHandle is the algorithm) which will immediately free the algorithm handle since the ref count will be zero. 

On the finalizer - the calling code will call Dispose on the algorithm right away, which disables the finalizer, but that doesn't really matter since the key's reference count will keep the handle alive until the key is disposed.

>> Honestly we could always set the parent handle and that'd be safer (though since we know we static-cache the rest of the algorithm handles it doesn't really matter).

Agree it would in theory be safer \ more correct to do that, although it just means extra (unnecessary) pinvokes to increment\decrement the ref count.